### ISSUE TITLE
Add .gitignore file

### ISSUE THREAD

**Issue by mik-dass** [fabric8-hdd/supervisor#8]
A .gitignore file is need for avoiding the unwanted files

**Comment by devang-gaur** [comment]
**TLDR;** 
Its not a good practice to include .gitignore file in upstream repositories. Its supposed to private by default. Every contibuter have their own content in their local .gitignore file.

**A bit long read**

Hey @anmolbabu and @dev-gaur , everyone have their own version of .gitignore files on their local repo of forks. It isn't usual for .gitignore files to be same and therefore, there's no point of maintaining a .gitignore files on the repo.

**Let me give a sample use case** :

if a contributer ( lets say @dev-gaur ) stores his linting result in a file for convenience and always runs  >"sh cico_run_eslint.sh > eslint_log.txt".
He also maintains a to-do list for himself, DG-TODO.md
Maybe he's a kishor kumar fan too and has kishore-da-songs/ directory in his local repo too. ( hypothecally ofcourse! ).  Then his .gitignore file would be like :

node_modules/
eslint_log.txt
DG-TODO.md
kishore-da-songs/

On the other hand,
if another guy ( let's say @anmolbabu ) likes to keep his repo clean and only has some log files and some random demo and test javascript files. Then his version of .gitignore would be different ofcourse :

node_modules/
test1.log.
demo1.js
dem_test.js

So, the point is that **_every contributor maintains their own .gitignore files' content as per their convenience._**

Even if we consider adding a .gitignore file, The only entry would be node_modules/ directory. And, if @dev-gaur fetches from the upstream repo and merges or rebases the fetched code, either unnecessary conflicts will occur in his local git setup or his original .gitignore content would be replaced by a lone "node_modules/" entry.

**_Whenever a contributor raises a PR, the reviewer is reponsible to check for unnecessary files and directories like node_modules/ and eslint.log added by mistake or ignorance._**

Thanks for reading :) I think this comment is long enough content for a short blog post or a quora answer!!!

@miteshvp  kindly correct me if am wrong here.

**Comment by miteshvp** [comment]
@dev-gaur actually that's a blog post and not a comment :)
Anyways, `.gitignore` can be part of main upstream if there is a common set of files that should be avoided. It makes things easy for any other developers who starts contributing later, does not commit any un-necessary files due to this `.gitignore`. As a reference, you can also look at https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/.gitignore which has a long list of ignore files, and works great for new developers or contributors.
As a side-note, gitignore in upstream should only contain things that affects every developer working on that repo. Developers can also add to gitignore files locally as well. Hope it helps. cc @mik-dass @anmolbabu

**Comment by anmolbabu** [comment]
@dev-gaur I would like to suggest you the following:
1. Try to keep your comments focused on subject and not write blog posts as above
2. If you are unsure whether something is a bug or not try to discuss it first in the team and then raise issues for that lets maintain the repo clean as much as possible.
3. Let's not have terms that reflect our attitude like "See for yourself", "Try for yourself" or even instructions to peers like "Merge PR x first then I'll send a PR to fix something else" etc.. like in https://github.com/fabric8-hdd/supervisor/issues/13

Let's keep our heads cool and try to maintain the dignity for each other within the team..

Please also see my responses on other issues..

@miteshvp Kindly provide your thoughts

**Comment by devang-gaur** [comment]
Hey @anmolbabu @miteshvp , thanks for correcting me. This is my first proper team repository. Prior to this, It was just my own projects and my dictatorship there :P . Sorry, I'll check my thoughts with more sincerety before getting them out into comments.

**Comment by anmolbabu** [comment]
Closing this as the PRs to fix this have been merged
